### DISCUSSION TITLE
Failed to create HTTPScaledObject, stuck in 'PendingCreation' state

### DISCUSSION THREAD

**Discussion by JimmyCater87** [kedacore/http-add-on#943] [General]
Issue : Failed to create HTTPScaledObject, stuck in 'PendingCreation' state 

Description:

1.  Followed the GIT Doc to install HTTP add-on in same namespace as Keda 
https://github.com/kedacore/http-add-on/blob/main/docs/install.md
Step: helm install http-add-on kedacore/keda-add-ons-http --namespace keda

2. Validate PODs status for Keda and HTTP add-on > all Pods are running

keda-add-ons-http-controller-manager  3/3     Running   0               5h35m
keda-add-ons-http-external-scaler      2/2     Running   0               12h
keda-add-ons-http-external-scaler       2/2     Running   0               12h
keda-add-ons-http-external-scaler       2/2     Running   0               12h
keda-add-ons-http-interceptor          2/2     Running   0               12h
keda-add-ons-http-interceptor         2/2     Running   0               12h
keda-add-ons-http-interceptor          2/2     Running   0               12h
keda-admission-webhooks              2/2     Running   1 (5h44m ago)   12h
keda-operator                        2/2     Running   1 (5h43m ago)   12h
keda-operator-metrics-apiserver     2/2     Running   1 (5h43m ago)   12h

3. 'HTTPScaledObject' scaler configuration 

kind: HTTPScaledObject
apiVersion: http.keda.sh/v1alpha1
metadata:
    name: http-scaler
spec:
    hosts:
    - 'host path'
    pathPrefixes:
    - 'prefix Path'
    targetPendingRequests: 10
    scaleTargetRef:
        deployment: httpbin
        service: httpbin
        port: 80
    replicas:
        min: 0
        max: 10
    scaledownPeriod: 300

---

Error Message in 'keda-add-ons-http-external-scaler' POD

“{"level":"error","ts":1710299145.4313052,"caller":"scaler/queue_pinger.go:146","msg":"getting request counts","error":"there isn't any valid interceptor endpoint","stacktrace":"main.(*queuePinger).fetchAndSaveCounts\n\tgithub.com/kedacore/http-add-on/scaler/queue_pinger.go:146\nmain.(*queuePinger).start\n\tgithub.com/kedacore/http-add-on/scaler/queue_pinger.go:108\nmain.main.func3\n\tgithub.com/kedacore/http-add-on/scaler/main.go:110\ngolang.org/x/sync/errgroup.(*Group).Go.func1\n\tgolang.org/x/sync@v0.6.0/errgroup/errgroup.go:78}"

---

Environment Details:
AKS Cluster 1.27.9 
Keda Version: 2.13.1 
HTTP Add-On version: 0.7.0

**Comment by vishu247** [comment]
@JimmyCater87 any solution for this issue?

**Comment by widdix123** [comment]
Has anyone found a solution for this ?

**Comment by cdavernas** [comment]
I solved it thanks to that [article](https://snowgoons.ro/posts/2024-07-03-ai-on-demand-with-kubernetes--part-2-/), by setting an ingress.

**Reply by widdix123** [reply]
Great . will have a look
